tests/jballerina-unit-test/src/test/resources/test-src/workers/worker-in-function-test.bal

```diff
@@ -26,3 +26,28 @@ function testSimpleWorkerVM(string msg) returns string {
     return wait first;
 }
 
+type Result record {
+    string a;
+    string b;
+};
+
+function fetch(string path) returns string {
+    return path;
+}
+
+function multiFetch(string pathA, string pathB) returns Result|error {
+    worker WA returns string {
+        return fetch(pathA);
+    }
+
+    worker WB returns string {
+            return fetch("pathB");
+    }
+
+    return wait {a: WA, b:WB};
+}
+
+public function testReturnWaitForAll() returns string|error {
+    Result res = check multiFetch("pathA", "pathB");
+    return res.a;
+}
```

Comment:
In the related issue, the sample code contain 
```
type Result record {
    string|error a;
    string|error b;
};
```
and return type of fetch function is `string|error`.
Workers also return `string|error`.
Shall we add a similar case as a test.